### PR TITLE
Rerender recipe 

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

File: .ci_support/migrations/hdf51140.yaml [hunk 1]
```diff
@@ -1,7 +0,0 @@
-__migrator:
-  build_number: 1
-  kind: version
-  migration_number: 1
-hdf5:
-- 1.14.0
-migrator_ts: 1675622758
```
Comment (jakirkham) [inline .ci_support/migrations/hdf51140.yaml]: The `hdf5` migrator was updated recently ( https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4594 ), which has caused it to be removed in some cases ( https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4594#issuecomment-1609308017 )

File: .ci_support/migrations/hdf51141.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+__migrator:
+  build_number: 1
+  kind: version
+  migration_number: 1
+hdf5:
+- 1.14.1
+migrator_ts: 1687165815.3104627
```
Comment (jakirkham) [inline .ci_support/migrations/hdf51141.yaml]: Have added the new `hdf5` migrator based on this commit ( https://github.com/conda-forge/conda-forge-pinning-feedstock/commit/77584f273c3fac007af0d55cd22b137ba7d02de9 )

File: .ci_support/linux_64_c_compiler_version10cuda_compilernvcccuda_compiler_version11.1cxx_compiler_version10fortran_compiler_version10mpimpichnumpy1.21python3.10.____cpython.yaml [hunk 3]
```diff
@@ -25,7 +25,7 @@ fortran_compiler:
 fortran_compiler_version:
 - '10'
 hdf5:
-- 1.14.0
+- 1.12.2
```
Comment (jakirkham) [inline .ci_support/linux_64_c_compiler_version10cuda_compilernvcccuda_compiler_version11.1cxx_compiler_version10fortran_compiler_version10mpimpichnumpy1.21python3.10.____cpython.yaml:28]: This is happening as the `hdf5` migrator was replaced (hence that file was dropped). Have attempted to fix this

**Review by jakirkham — COMMENTED** [review]
Hey Jan 👋

Pushed a change that should help with the `hdf5` migration situation. Added some context below.

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

**Pull request merged**